utils/aws_api.py

```diff
@@ -0,0 +1,369 @@
+                    has_owner = True
+                    break
+        return has_owner
+
+    def custom_s3_filter(self, account, s3, buckets):
+        type = 's3 bucket'
+        unfiltered_buckets = []
+        for b in buckets:
+            if self.resource_has_special_name(account, type, b):
+                continue
+            # ignore buckets with special tags
+            try:
+                tags = s3.get_bucket_tagging(Bucket=b)['TagSet']
+                if self.resource_has_special_tags(account, type, b, tags):
+                    continue
+                unfiltered_buckets.append(b)
+            except Exception as e:
+                if 'NoSuchTagSet' in e.message:
+                    unfiltered_buckets.append(b)
+        return unfiltered_buckets
+
+    def custom_sqs_filter(self, account, sqs, queues):
+        type = 'sqs queue'
+        unfiltered_queues = []
+        for q in queues:
+            if self.resource_has_special_name(account, type, q):
+                continue
+            # ignore queues with special tags
+            try:
+                tags = sqs.list_queue_tags(QueueUrl=q)['Tags']
+                if self.resource_has_special_tags(account, type, q, tags):
+                    continue
+                unfiltered_queues.append(q)
+            except KeyError as e:
+                if 'Tags' in e.message:
+                    unfiltered_queues.append(q)
+        return unfiltered_queues
+
+    def custom_dynamodb_filter(self, account, session, dynamodb, tables):
+        type = 'dynamodb table'
+        dynamodb_resource = session.resource('dynamodb')
+        unfiltered_tables = []
+        for t in tables:
+            if self.resource_has_special_name(account, type, t):
+                continue
+            # ignore queues with special tags
+            try:
+                table_arn = dynamodb_resource.Table(t).table_arn
+                tags = dynamodb.list_tags_of_resource(
+                    ResourceArn=table_arn)['Tags']
+                if self.resource_has_special_tags(account, type, t, tags):
+                    continue
+                unfiltered_tables.append(t)
+            except KeyError as e:
+                if 'Tags' in e.message:
+                    unfiltered_tables.append(t)
+        return unfiltered_tables
+
+    def custom_rds_filter(self, account, rds, instances):
+        type = 'rds instance'
+        unfiltered_instances = []
+        for i in instances:
+            if self.resource_has_special_name(account, type, i):
+                continue
+            # ignore queues with special tags
+            try:
+                instance = rds.describe_db_instances(DBInstanceIdentifier=i)
+                instance_arn = instance['DBInstances'][0]['DBInstanceArn']
+                tags = rds.list_tags_for_resource(
+                    ResourceName=instance_arn)['TagList']
+                if self.resource_has_special_tags(account, type, i, tags):
```

Comment:
again a bit of style, but instead of adding the continue, why not just do:

```python
if not <expr>:
    bla
```

It's much cleaner than:

```
if <expr>:
   continue
bla
```

Reply:
just to clarify in the cases where you have a big chunk of code below the if and you don't want to indent it all, the if continue makes sense, but in this case it's pointless, because it's just one statement in the if, with no additional indentation. So I'm not suggesting you change the if continue's above, only this one

Reply:
fixed